### PR TITLE
fix indigo moveit_kinematics Eigen3 dependency

### PR DESCRIPTION
**Pull request by YuehChuan**
### Description
#469    
in folder ~/moveit/moveit_kinematics/CMakeLists.txt
comment out  ${EIGEN3_INCLUDE_DIRS}
```bash=
catkin_package(
  LIBRARIES
  INCLUDE_DIRS
#    ${EIGEN3_INCLUDE_DIRS}
```
https://github.com/YuehChuan/moveit/blob/indigo-devel/moveit_kinematics/CMakeLists.txt#L35

in folder ~/moveit/moveit_kinematics/package.xml ,add cmake modules
```xml=
<build_depend>cmake_modules</build_depend>
<run_depend>cmake_modules</run_depend>
```
https://github.com/YuehChuan/moveit/blob/indigo-devel/moveit_kinematics/package.xml#L26

### Checklist
- [ ] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Optional: Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Optional: Decide if this should be cherry-picked to other current ROS branches (Indigo, Jade, Kinetic)

Thank you!

### CODE REVIEW

File: moveit_kinematics/CMakeLists.txt [hunk 1]
```diff
@@ -32,12 +32,13 @@ set(THIS_PACKAGE_INCLUDE_DIRS
 catkin_package(
   LIBRARIES
   INCLUDE_DIRS
-    ${EIGEN3_INCLUDE_DIRS}
+#    ${EIGEN3_INCLUDE_DIRS}
```
Comment (de-vri-es) [inline moveit_kinematics/CMakeLists.txt]: Could you remove the line instead of commenting it out?

File: moveit_kinematics/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(moveit_kinematics)
-find_package(catkin REQUIRED)
+find_package(catkin REQUIRED cmake_modules)
```
Comment (de-vri-es) [inline moveit_kinematics/CMakeLists.txt]: Why is this needed? I seem to recall that using `cmake_modules` for Eigen is deprecated these days. Besides that I don't see other changes to match with this. If this was needed in order to compile on your system there may be more wrong.

Comment (YuehChuan) [inline moveit_kinematics/CMakeLists.txt]: Uhha, you are right. It's for compiling eigen3, and it is deprecated.  I'm still using the old style.
Already remove, thanks for let me know this.

File: moveit_ros/planning/kdl_kinematics_plugin/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+set(MOVEIT_LIB_NAME moveit_kdl_kinematics_plugin)
```
Comment (davetcoleman) [inline moveit_ros/planning/kdl_kinematics_plugin/CMakeLists.txt]: none of these KDL files should be in this pull request-  it a mistake right?

Comment (YuehChuan) [inline moveit_ros/planning/kdl_kinematics_plugin/CMakeLists.txt]: hey @davetcoleman  Dave thanks for reviewing! yep you are right, there is noting to do with KDL plugin.  I forgot why i added it.  Tones of moveIt stuff I don't know. Is kdl also deprecated or is self-contain in newest version? maybe be the habit from gazebo.

We are try moveIt! on a new collaborative arm robot  **techman robot** made from taiwan
http://tm-robot.com/
Now just made rviz could connect to real robot and gazebo model;  And there is some challenge in motion plannning like RRTconnectkConfig; some poses the robot could not reach or the plannig path it's strang. Is there some reference for add constrain or specified the motion?  
here are repo's 
my lab: https://github.com/ISCI-NCTU/isci_ros_tm5
official: https://github.com/kentsai0319/techman_robot
Just in case you guys get interest and have free time to play, hehehe.  
Anyway, this is not relate to this PR.  Thanks for review! :smile: 

moveit_configuration

File: moveit_kinematics/CMakeLists.txt [hunk 4]
```diff
@@ -32,12 +32,12 @@ set(THIS_PACKAGE_INCLUDE_DIRS
 catkin_package(
   LIBRARIES
   INCLUDE_DIRS
-    ${EIGEN3_INCLUDE_DIRS}
     ${THIS_PACKAGE_INCLUDE_DIRS}
   CATKIN_DEPENDS
     pluginlib
     moveit_core
     moveit_ros_planning
+    DEPENDS Eigen3
```
Comment (v4hn) [inline moveit_kinematics/CMakeLists.txt]: This doesn't work.
`pkg_search_module(EIGEN3 REQUIRED eigen3)` defines a variable `EIGEN3_INCLUDE_DIRS`, but
the `DEPENDS Eigen3` makes catkin look for `Eigen3_INCLUDE_DIRS`.

Please change this to (including the newline+indention)
```
DEPENDS
  EIGEN3
```

File: moveit_ros/planning/kdl_kinematics_plugin/CMakeLists.txt (hunk 3 above)
Comment (davetcoleman) [inline moveit_ros/planning/kdl_kinematics_plugin/CMakeLists.txt]: @YuehChuan please keep discussion to original topic in this thread, thank you!

File: moveit_kinematics/CMakeLists.txt (hunk 4 above)
Comment (130s) [inline moveit_kinematics/CMakeLists.txt]: Ping @YuehChuan :)

Comment (YuehChuan) [inline moveit_kinematics/CMakeLists.txt]: @130s 64 bytes from YuehCHuan: icmp_seq=31 ttl=64 time=0.189 ms

Comment (YuehChuan) [inline moveit_kinematics/CMakeLists.txt]: what's ping mean

Comment (YuehChuan) [inline moveit_kinematics/CMakeLists.txt]: OH i didn't submit mine review 10 days ago seems they were pending, got it, thanks @130s

Comment (v4hn) [inline moveit_kinematics/CMakeLists.txt]: @YuehChuan See "Ping" here: http://moveit.ros.org/documentation/contributing/pullrequests/

Comment (v4hn) [inline moveit_kinematics/CMakeLists.txt]: Passing on Eigen dependencies in ROS is a bit of a curiosity really. It was/(is?) broken in many places and nobody ever notices, because every dependent project still includes it *somehow* either directly or through different dependency.
It doesn't surprise me that travis succeeds even when your fix probably does not work.
This is only made worse by the fact that until a couple of months ago only Debian's version of eigen3 provided the variable `EIGEN3_INCLUDE_DIRS` (notice the trailing `S`) and everyone else only set `EIGEN3_INCLUDE_DIR` (no `S`). Anyway..

> But , I didn't get it.

Variables in cmake are case-sensitive.
`catkin_package`'s `DEPENDS <package>` field invokes some logic that automatically adds the contents of the variables `<package>_INCLUDE_DIRS` and `<package>_LIBRARIES` to projects that find_package this catkin package. See [here](https://github.com/ros/catkin/blob/kinetic-devel/cmake/catkin_package.cmake#L168).

So with the code you currently propose `catkin_package` will look at the variables `Eigen3_INCLUDE_DIRS` and `Eigen3_LIBRARIES`.
eigen is header-only, so the latter doesn't matter.
But the variable `Eigen3_INCLUDE_DIRS` doesn't exist either. It should be looking for `EIGEN3_INCLUDE_DIRS` instead.

The newline is just a matter of taste and imho fits better with the rest of the CMakeLists.txt.

Comment (YuehChuan) [inline moveit_kinematics/CMakeLists.txt]: @v4hn thanks, you are so kind:)
Great insite  for explaning eigen so detail.
I'm sorry for reply this late. Your advices really help me learn a lot. 
Most people write CmakeLists.txt using template without understanding
what has been happened inside actually. Yes, add Eigen dependencies is out of curiosity.
To tell the truth I may not quite understand the meaning before you give the explanation.

Anyway thank you. Ping me more often please, great teacher. I'll reply it soon next time.:p
(Mailbox full of github notifications...)

Comment (v4hn) [inline moveit_kinematics/CMakeLists.txt]: Could you please address my feedback then? :)

Comment (YuehChuan) [inline moveit_kinematics/CMakeLists.txt]: @v4hn Thank you! Changed.

**Comment by v4hn** [comment]
> Do we want to keep using pkg-config for Eigen, or depend on Eigen3Config.cmake (which sadly exports EIGEN3_* variables, though that does mean it is compatible with the current cmake code of moveit).

We use both ways in MoveIt at the moment.
pkg-config is more backward compatible because until recently upstream eigen3 didn't provide `Eigen3Config.cmake` in a release.
Other than that it doesn't really matter.

**Comment by v4hn** [comment]
cherry-picked to j/k

**Comment by YuehChuan** [comment]
@v4hn Thanks for those feed!
